client/src/components/campaignCreatorComponents/CampaignCreator.tsx

```diff
@@ -36,8 +43,81 @@ const CampaignCreator = ({ campaignSettings, setCampaignSettings }: CampaignCrea
+                    const fight_year: number | null = selectedCampaign[fight_key];
+                    if (fight_year !== null) {
+                        timeline[fight_year].push(`NE - ${selection} lvl ${level}`);
+                    }
+                });
+            });
+            return;
+        }
+
+        selectedCampaign[nodeKey].forEach((selection: string) => {
+            const query: Record<string, Record<string, string[]>> = campaignOptionsData.timeline[typeKey];
+            if (query.hasOwnProperty(selection)) {
+                Object.keys(query[selection]).forEach((yearKey: string) => {
+                    query[selection][Number(yearKey)].forEach((yearData: string) => {
+                        timeline[Number(yearKey)].push(yearData);
+                    });
+                });
+            }
+        });
+    };
+
     const handleCreateTimeline = (event: React.FormEvent<HTMLFormElement>) => {
         event.preventDefault();
+        if (settlementName.length < 2) {
+            // TODO: Add Toast popup with warning
+            return;
+        }
+        // Populate Timeline
+        const maximumYears: number = 40;
+        const timeline: Record<number, string[]> = Array.from({ length: maximumYears }, (_, index) => index + 1).reduce(
+            (obj, key) => ({ ...obj, [key]: [] }),
+            {},
+        );
+        addToTimeline(timeline, 'node_quarry_1', 'quarries');
+        addToTimeline(timeline, 'node_quarry_2', 'quarries');
+        addToTimeline(timeline, 'node_quarry_3', 'quarries');
+        addToTimeline(timeline, 'node_quarry_4', 'quarries');
+        addToTimeline(timeline, 'pillars', 'pillars');
+        addToTimeline(timeline, 'encounters', 'encounters');
+        addToTimeline(timeline, 'wanderers', 'wanderers');
+        addToTimeline(timeline, 'node_nemesis_1', 'nemesis');
+        addToTimeline(timeline, 'node_nemesis_2', 'nemesis');
+        addToTimeline(timeline, 'node_nemesis_3', 'nemesis');
+
+        // Add Core and Finale to Timeline
+        if (selectedCampaign.core_fight_year !== null && selectedCampaign.node_core !== null) {
+            timeline[selectedCampaign.core_fight_year].push(`NE - ${selectedCampaign.node_core}`);
+        }
+        if (selectedCampaign.finale_fight_year !== null && selectedCampaign.node_finale !== null) {
+            timeline[selectedCampaign.finale_fight_year].push(`NE - ${selectedCampaign.node_finale}`);
+        }
+        // Add default events to timeline
+        Object.keys(selectedCampaign.default_timeline).forEach((yearKey: string) => {
+            selectedCampaign.default_timeline[Number(yearKey)].forEach((yearData: string) => {
+                timeline[Number(yearKey)].push(yearData);
+            });
+        });
+        // console.log(timeline);
+        const updatedCampaign = { ...selectedCampaign };
+        updatedCampaign.timeline = { ...timeline };
```

Comment:
A lot of this is just semi-functional
Not in like the it doesn't work way
But in that objects are being cloned with spread, but then `timeline` is like it's own thing 
The patterns are sprinkled in but the use is inconsistent and confusing